### PR TITLE
IsInternalDomain - added new script

### PR DESCRIPTION
**Pull request by abaumgarten**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/26223

## Description
New python script to check if a domain or a sub domain is part of a given domain

## Minimum version of Demisto
- [x] 5.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            'Domain.Name': element,
+            'Domain.IsInternal': True if is_in_domain else False
+        })
+    return CommandResults(outputs=context_entry)
+
+
+def main():
+    args = demisto.args()
+    domain_name = args.get('domainName')
+    domain_to_check = argToList(args.get('domainToCheck'))
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py:34]: How about `possible_sub_domains` or `sub_domains_to_check`?

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    Args:
+        domain_name: main domain
+        domain_to_check: list of domains or sub domains that should be checked
+
+    Returns:
+        for each domain for the list an entry with True / False if it is in the domain or not
+    """
+    context_entry = []
+    for element in domain_to_check:
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py:18]: let's try to find a better name than `element`

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        is_in_domain = False
+        # split by domain name
+        domain_to_check_prefix = element.split(domain_name)[0]
+        if domain_to_check_prefix + domain_name == element:
+            is_in_domain = True
+        context_entry.append({
+            'Domain.Name': element,
+            'Domain.IsInternal': True if is_in_domain else False
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py:26]:
```suggestion
            'Domain.IsInternal': is_in_domain
```

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    Returns:
+        for each domain for the list an entry with True / False if it is in the domain or not
+    """
+    context_entry = []
+    for element in domain_to_check:
+        is_in_domain = False
+        # split by domain name
+        domain_to_check_prefix = element.split(domain_name)[0]
+        if domain_to_check_prefix + domain_name == element:
+            is_in_domain = True
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py:23]:
```suggestion
        is_in_domain = (domain_to_check_prefix + domain_name == element)
```

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        domain_to_check: list of domains or sub domains that should be checked
+
+    Returns:
+        for each domain for the list an entry with True / False if it is in the domain or not
+    """
+    context_entry = []
+    for element in domain_to_check:
+        is_in_domain = False
+        # split by domain name
+        domain_to_check_prefix = element.split(domain_name)[0]
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py:21]: why not just use `in`? what am I missing?

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
+"""
+Script to check if a domain or a sub domain is part of a given domain
+"""
+from CommonServerPython import *
+
+
+def check_in_domain(domain_name: str, domain_to_check: list) -> CommandResults:
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py:7]: let's give this method a better name

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.yml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  description: The domain name that the sub domain should be in
+  isArray: false
+  name: domainName
+  required: true
+  secret: false
+- auto: PREDEFINED
+  default: true
+  description: A comma-separated list of domains to check.
+  isArray: true
+  name: domainToCheck
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.yml:12]: let's change to Python style naming, here and in other arguments

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.yml [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  id: IsInternalDomainName
+  version: -1
+enabled: false
+name: IsInternalDomainName
+outputs:
+- contextPath: Domain.Name
+  description: The domain that was checked
+  type: String
+- contextPath: Domain.IsInternal
+  description: "True if sub domain is part of a domain. \nFalse otherwise"
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.yml:26]: Shouldn't this also include the domain it was checked against?
@David-BMS WDYT?

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName_test.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+    When:
+        - running the script
+
+    Then:
+        - returns for each sub domain if it is a sub domain of given domainName or not
+
+    """
+    result = check_in_domain(domain_name, domain_to_check)
+    for i, element in enumerate(result.outputs):
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName_test.py:34]: let's rename `i, element`

File: Packs/CommonScripts/pack_metadata.json [hunk 10]
```diff
@@ -2,7 +2,7 @@
     "name": "Common Scripts",
     "description": "Frequently used scripts pack.",
     "support": "xsoar",
-    "currentVersion": "1.2.53",
+    "currentVersion": "1.3.0",
```
Comment (guykeller) [inline Packs/CommonScripts/pack_metadata.json:5]: Why such a bump to the version?

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.yml (hunk 8 above)
Comment (abaumgarten) [reply Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.yml]: Done as David wrote in the issue

**Comment by ShirleyDenkberg** [comment]
@guykeller Review is done.

**Review by guykeller — APPROVED** [review]

**Comment by guykeller** [comment]
@abaumgarten Added some comments, please do two things before you merge:
1. Review the last comments and change where needed.
2. @David-BMS Could you please test the script before we merge it?

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+def check_sub_domains_in_domain(domain_name: str, sub_domains_to_check: list) -> CommandResults:
+    """
+
+    Args:
+        domain_name: main domain
+        sub_domains_to_check: list of domains or sub domains that should be checked
+
+    Returns:
+        for each domain for the list an entry with True / False if it is in the domain or not
```
[the remark this answers is not in the document]
Comment (abaumgarten) [reply Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py]: Done

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py [hunk 12]
```diff
@@ -0,0 +1,47 @@
+"""
+Script to check if a domain or a sub domain is part of a given domain
+"""
+from CommonServerPython import *
+
+LIMIT_OF_TABLE = 10
```
[the remark this answers is not in the document]
Comment (abaumgarten) [reply Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py]: Done

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+    table = tableToMarkdown("", markdown[:LIMIT_OF_TABLE], headers)
+    return CommandResults(outputs={'IsInternalDomain': context_entry}, readable_output=table)
+
+
+def main():
+    args = demisto.args()
+    domain_name = args.get('domain_name')
+    sub_domains_to_check = argToList(args.get('domain_to_check'))
+    if len(sub_domains_to_check) == 0:
+        return_error("IsInternalDomainName has to get at least one domainToCheck")
```
[the remark this answers is not in the document]
Comment (abaumgarten) [reply Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py]: Done

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+        2. readable_output (markdown table) : contains first 10 entries with the above headers:
+           ["DomainToTest", "DomainToCompare", "IsInternal"]
+    """
+    context_entry = []
+    markdown = []
+    headers = ["DomainToTest", "DomainToCompare", "IsInternal"]
+    for sub_domain in sub_domains_to_check:
+        is_in_domain = False
+        for main_domain in domains_to_compare:
+            if main_domain in sub_domain:
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py:34]: let's use `any()` here

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+
+
+def main():
+    args = demisto.args()
+    domains_to_compare = argToList(args.get('domains_to_compare'))
+    if len(domains_to_compare) == 0:
+        return_error("Error: please specify at least one possible main domain to compare with.")
+    sub_domains_to_check = argToList(args.get('domain_to_check'))
+    if len(sub_domains_to_check) == 0:
+        return_error("Error: please specify at least one possible sub domain to check.")
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py:58]: Let's add these lines + lines 54-55 to a helper function called `validate_args` or something like that

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py (hunk 14 above)
Comment (abaumgarten) [reply Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py]: Done

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py (hunk 15 above)
Comment (abaumgarten) [reply Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py]: Done

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py (hunk 14 above)
Comment (abaumgarten) [reply Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py]: Done

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py (hunk 15 above)
Comment (abaumgarten) [reply Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.py]: Done

**Review by guykeller — APPROVED** [review]

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.yml [hunk 16]
```diff
@@ -0,0 +1,43 @@
+args:
+- auto: PREDEFINED
+  default: true
+  description: A comma-separated list of main - domains.
```
Comment (David-BMS) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.yml:4]: @ShirleyDenkberg we want to emphasize that these are the domains we want to check (for example apps.google.com)

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.yml [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+  isArray: true
+  name: domains_to_compare
+  required: true
+  secret: false
+- auto: PREDEFINED
+  default: true
+  description: A comma-separated list of subdomains.
```
Comment (David-BMS) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.yml:11]: @ShirleyDenkberg These are the domains we want to test that are a part of the previous argument.

File: Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.yml [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+  isArray: true
+  name: domain_to_check
+  required: true
+  secret: false
+comment: Checks if a subdomain is located in at least one of the given domains to
+  compare with.
```
Comment (David-BMS) [inline Packs/CommonScripts/Scripts/IsInternalDomainName/IsInternalDomainName.yml:17]: @ShirleyDenkberg For example, is apps.paloaltonetworks.com is a part of paloaltonetworks.com
The script can accept multiple values for both arguments and will iterate through each of the domains we want to check if it is in the list of the domains we want to compare with. As long as the tested domain is in one of the compared domains the is internal will be set as true. For example our domains to check are apps.paloaltonetworks.com,and apps.paloaltonetworks.bla and the domains_to_compare are paloaltonetworks.com and demisto.com
The result for apps.paloaltonetworks.com will be true as its a part of paloaltonetworks.com and the result for apps.paloaltonetworks.bla will be false as its nether a part of paloaltonetworks.com or demisto.com

**Comment by David-BMS** [comment]
> @abaumgarten Added some comments, please do two things before you merge:
> 
> 1. Review the last comments and change where needed.
> 2. @David-BMS Could you please test the script before we merge it?

Script looks great
I added some documentation notes for @ShirleyDenkberg

File: Packs/CommonScripts/ReleaseNotes/1_2_56.md [hunk 19]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### New: IsInternalDomainName
+- Checks if a subdomain is located in the specified domain.
```
Comment (guykeller) [inline Packs/CommonScripts/ReleaseNotes/1_2_56.md:4]: @ShirleyDenkberg I think we should reconsider if this is a good phrasing, I don't think it accurately describes what the script does

**Comment by ShirleyDenkberg** [comment]
@David-BMS @abaumgarten Implemented David's comments.

**Pull request merged**